### PR TITLE
Create contests page

### PR DESCRIPTION
**Pull request by miguelrodrigues22**

### CODE REVIEW

File: pages/contests/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import Head from "next/head";
+import Image from "next/image";
+import styles from "../../styles/Home.module.css";
+import { useState } from "react";
+
+function RenderContest({ contest }) {
```
Comment (JoaoCVerissimo) [inline pages/contests/index.js:6]: u can deconstruct props like:
function renderContest(props) {
const {name, description} = props
...
}

File: pages/contests/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                        {contest.date_end.slice(0, 10)}
+                    </p>
+                </div>
+            </div>
+        </>
+    )
+}
+
+
+export default function Home({ contestList }) {
```
Comment (JoaoCVerissimo) [inline pages/contests/index.js:27]: same thing here, we don't need to always do styles.<something> if we deconstruct first

File: pages/index.js [hunk 3]
```diff
@@ -1,24 +1,129 @@
-import Head from 'next/head'
-import Image from 'next/image'
-import styles from '../styles/Home.module.css'
+import Head from "next/head";
+import Image from "next/image";
+import styles from "../styles/Home.module.css";
+import { useState } from "react";
+
+const DECODE_PREFIX = "data:image/png;base64,";
+const API = "http://127.0.0.1:8000/photo/";
```
Comment (JoaoCVerissimo) [inline pages/index.js:7]: this is a .env variable

File: pages/contests/index.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    return (
+        <div className={styles.container}>
```
Comment (JoaoCVerissimo) [inline pages/contests/index.js:30]: change styles for tailwind classes.

File: pages/index.js [hunk 5]
```diff
@@ -1,24 +1,129 @@
...
+async function getSubmissionListWithContest() {
+  return fetch(API + "submissions/", {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+    },
+  }).then((data) => data.json());
+}
+
+async function getUserList() {
```
Comment (JoaoCVerissimo) [inline pages/index.js:18]: maybe have a single file to make requests only, export them, and import here to use

File: pages/contests/index.js (hunk 2 above)
Comment (miguelrodrigues22) [reply pages/contests/index.js]: done

File: pages/contests/index.js (hunk 1 above)
Comment (miguelrodrigues22) [reply pages/contests/index.js]: done

File: pages/index.js (hunk 5 above)
Comment (miguelrodrigues22) [reply pages/index.js:18]: this was in André's PR, maybe he already did it?

File: .babelrc [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (hugo-serra) [inline .babelrc:1]: This is not needed

File: .eslintrc.json [hunk 7]
```diff
@@ -1,3 +1,3 @@
 {
-  "extends": "next/core-web-vitals"
+  "extends": ["next/babel","next/core-web-vitals"]
```
Comment (hugo-serra) [inline .eslintrc.json:2]: Babel not needed

File: pages/contests/index.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
+import Head from "next/head";
+import Image from "next/image";
+import styles from "../../styles/Home.module.css";
```
Comment (hugo-serra) [inline pages/contests/index.js:3]: Let's add the style to _app.js or to document (see nextsjs docs)

File: pages/index.js [hunk 9]
```diff
@@ -1,24 +1,129 @@
...
-import styles from '../styles/Home.module.css'
+import Head from "next/head";
+import Image from "next/image";
+import styles from "../styles/Home.module.css";
+import { useState } from "react";
+
+const DECODE_PREFIX = "data:image/png;base64,";
+const API = "http://127.0.0.1:8000/photo/";
+
+async function getSubmissionListWithContest() {
```
Comment (hugo-serra) [inline pages/index.js:9]: This is supposed to be in a service

File: pages/index.js (hunk 5 above)
Comment (hugo-serra) [reply pages/index.js:18]: Remember to pull when @andrevieira11 fixes this

File: styles/Home.module.css [hunk 10]
```diff
@@ -12,6 +12,16 @@
   align-items: center;
 }
 
+.contest_container {
+  min-height: 100vh;
```
Comment (hugo-serra) [inline styles/Home.module.css:16]: We shouldn't be using custom css, just use whatever tailwind provides

**Review by hugo-serra — CHANGES_REQUESTED** [review]

File: .babelrc (hunk 6 above)
Comment (miguelrodrigues22) [reply .babelrc:1]: done

File: .eslintrc.json (hunk 7 above)
Comment (miguelrodrigues22) [reply .eslintrc.json:2]: done

File: styles/Home.module.css (hunk 10 above)
Comment (miguelrodrigues22) [reply styles/Home.module.css:16]: not beeing used anymore

File: pages/index.js (hunk 5 above)
Comment (miguelrodrigues22) [reply pages/index.js]: done

File: pages/contests/index.js (hunk 8 above)
Comment (miguelrodrigues22) [reply pages/contests/index.js:3]: done

File: pages/index.js (hunk 9 above)
Comment (miguelrodrigues22) [reply pages/index.js]: done

**Review by JoaoCVerissimo — APPROVED** [review]

File: .env [hunk 11]
```diff
@@ -0,0 +1,9 @@
+NEXT_PUBLIC_API_URL="http://127.0.0.1:8000/api/"
+GOOGLE_CLIENT_ID=185796748522-q2t69jsidgopsqji9b7eujgqk945up7j.apps.googleusercontent.com
```
Comment (hugo-serra) [inline .env:2]: Never submit secrets to git. @JoaoCVerissimo when you're available help out here

File: dist/output.css [hunk 12]
```diff
@@ -0,0 +1,801 @@
+/*
```
Comment (hugo-serra) [inline dist/output.css:1]: This file should only include imports from tailwind not tailwind itself

File: pages/contest/[id].js [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+                description={description}
+              />
+            </div>
+          ))}
+        </div>
+        <div>
+          <h4>Select Image</h4>
+          <input type="file" name="myImage" onChange={loadImage} />
+
+          {createObjectURL && (
```
Comment (hugo-serra) [inline pages/contest/[id].js:64]: what would the value of this createObjectUrl be?

File: pages/contest/[id]/submission/[submissionId].js [hunk 14]
```diff
@@ -0,0 +1,21 @@
+export default function submission({ newID, newID2 }) {
+  return (
+    <div>
+      <p>{newID}</p>
```
Comment (hugo-serra) [inline pages/contest/[id]/submission/[submissionId].js:4]: What are newIDs? What are they IDs of?

File: services/reventService.js [hunk 15]
```diff
@@ -0,0 +1,52 @@
+const { NEXT_PUBLIC_API_URL } = process.env;
+const GET = "GET";
+const POST = "POST";
+
+export async function fetchEndpoint(endpoint, method, ...args) {
+  if (method == GET) {
```
Comment (hugo-serra) [inline services/reventService.js:6]: the only thing that is different when we have GET or POST is the body, so we should only change the body inside the if, and if there is a body add it to the message. Then, make the request as usual

**Review by hugo-serra — CHANGES_REQUESTED** [review]

File: pages/contest/[id].js [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+import { imageToUrl } from "@/services/imageDecoderService.js";
+import {
+  getSubmissionsFromContest,
+  getUserList,
+  postSubmission,
+} from "@/services/reventService.js";
+import Head from "next/head";
+import Router from "next/router";
+import { useState } from "react";
+import styles from "../../styles/Home.module.css";
```
[the remark this answers is not in the document]
Comment (miguelrodrigues22) [reply pages/contest/[id].js:11]: André solved it

File: pages/contest/[id]/submission/[submissionId].js (hunk 14 above)
Comment (miguelrodrigues22) [reply pages/contest/[id]/submission/[submissionId].js:4]: André's code

**Review by hugo-serra — APPROVED** [review]

**Pull request merged**